### ISSUE TITLE
[no-restricted-imports] extend to support type-only imports

### ISSUE THREAD

**Issue by yoyo930021** [typescript-eslint/typescript-eslint#2661]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

We need to limit only type import some packages sometimes.
For example:
When using package in user project runtimes, Like vetur, ts-loader.

## Rule Details
This rule limit only type import some package within your codebase.

### Options
```typescript
type Options = {
  packages: Array<{ name: string, reason: string }>
};
```

### Example
Incorrect code for `{ packages: [{ name: 'typescript', reason: 'runtime dependency' }] }`
```typescript
import ts from 'typescript'
import * as ts from 'typescript'
import { findPackageConfig } from 'typescript'
```

correct code for `{ packages: [{ name: 'typescript', reason: 'runtime dependency' }] }`
```typescript
import type ts from 'typescript'
import type * as ts from 'typescript'
import type { findPackageConfig } from 'typescript'
```

**Comment by bradzacher** [comment]
You can use `no-restricted-syntax` to do this pretty easily.

Best case could do an extension rule for [no-restricted-imports] (https://eslint.org/docs/rules/no-restricted-imports)

It might be better instead to document the `importKind` property in estree itself and then support this officially in the core rule because its such a minor change which is not TS specific (its also used by babel to denote type imports in flow)

https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/types/src/ts-estree.ts#L994

**Comment by yoyo930021** [comment]
Ok, I will open a issue to eslint repo.
Thanks.

**Comment by yoyo930021** [comment]
They don't to add this rule in eslint core rules.
https://github.com/eslint/eslint/issues/13758#issuecomment-709353258

Would you be interested in this project?
Or I can implement the rules myself and put them somewhere.

**Comment by darkclouder** [comment]
I would also love to have this automatically.
Especially if you have more than a single class in a file, it can quickly get frustrating to check and also do this manually.
So eslint should automatically check which imports are only used as a type and optimise imports for that.

**Comment by bradzacher** [comment]
@darkclouder this issue is specifically about restricting a module to only be imported via types.
If you're looking for a rule to enforce that things only imported as types are specified as type-only imports, consider instead [`consistent-type-imports`](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/consistent-type-imports.md)

**Comment by bradzacher** [comment]
looking at the discussion in the related eslint issue, this is going to have to be an extension rule.
happy to accept a PR!
